src/main/java/org/spongepowered/api/event/inventory/InventoryOpenEvent.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.inventory;
+
+import org.spongepowered.api.util.event.Cancellable;
+
+/**
+ * Fired when a viewer opens an inventory.
+ */
+public interface InventoryOpenEvent extends ViewerEvent, Cancellable {
```

Comment:
@simon816 It's certainly possible to do.  Graphically, it looks a little weird(their inventory opens then suddenly closes), so it's not exactly ideal, but it works.  The biggest issue I see, however, is the potential security issues that are created by naively closing the window.  When you close most inventories, you assume that the player(or really, the client) is unable to interact with the inventory.  Hacked clients, however, could still modify the inventory as no management of whether or not they have it open exists.

If you want to implement this "correctly" and in a manner that's consistent. you'd need to add in complete inventory tracking, presumably by interpreting the achievement packet as an openwindow packet and storing the state of whether they actually have their inventory open.